web-platform/website/src/components/pages/account-settings/EditAccountEmailView.tsx

```diff
@@ -0,0 +1,278 @@
+import { FirebaseError } from "@firebase/app";
+import { useEffect, useRef, useState } from "react";
+import {
+  Col,
+  Container,
+  Card,
+  Form,
+  FormFeedback,
+  FormGroup,
+  Input,
+  Label,
+  Row,
+  Button
+} from "reactstrap";
+
+import { style } from "typestyle";
+import {
+  LoginFormValidationResult,
+  validateLoginForm,
+} from "../login/LoginFormValidator";
+import { EmailChangedView } from './EmailChangedView'
+import { Strings } from "../../../resources/Strings";
+import { useAuthentication } from "../../../services/AuthenticationService";
+import { Colors, Spacing } from "../../../styles";
+import { DisabledProductButton, ProductButton, TertiaryButton, TransparentButton } from "../../../styles/Buttons";
+import { ContainerStyles } from "../../../styles/ContainerStyles";
+import { FontSizeRaw, Fonts } from "../../../styles/Fonts";
+import { getFirebaseErrorMessage } from "../../../utils/FirebaseErrors";
+import {
+  AccountSettingsState,
+  useAccountSettingsDataContext,
+} from "./AccountSettingsDataContext";
+
+
+const strings = Strings.components.pages.accountSettings.editEmailAccount;
+const firebaseStings = Strings.utils.firebaseError.errorMessages
+
+export function EditAccountEmailView() {
+  const [confirmationView, setConfirmationView] = useState(false)
+  const [email, setEmail] = useState("");
+  const [isDisabled, setDisabled] = useState(true);
+  const [password, setPassword] = useState("");
+  const [passwordVisible, setPasswordVisbile] = useState(false);
+  const [eyeIconVisible, setVisibility] = useState(false)
```

Comment:
This can be inferred via passwordVisible.

Reply:
This should remain since it changes based on a different set of user actions.